trace-utils/src/span_v04/trace_utils.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright 2021-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+//! Trace-utils functionalities implementation for tinybytes based spans
+
+use std::collections::HashMap;
+use tinybytes::BytesString;
+
+use super::Span;
+
+/// Span metric the mini agent must set for the backend to recognize top level span
+const TOP_LEVEL_KEY: &str = "_top_level";
+/// Span metric the tracer sets to denote a top level span
+const TRACER_TOP_LEVEL_KEY: &str = "_dd.top_level";
+const MEASURED_KEY: &str = "_dd.measured";
+const PARTIAL_VERSION_KEY: &str = "_dd.partial_version";
+
+fn set_top_level_span(span: &mut Span, is_top_level: bool) {
+    if !is_top_level {
+        if span.metrics.contains_key(TOP_LEVEL_KEY) {
+            span.metrics.remove(TOP_LEVEL_KEY);
+        }
+        return;
+    }
+    span.metrics.insert(TOP_LEVEL_KEY.into(), 1.0);
+}
```

Comment:
/improvement

you can simplify

```suggestion
   if is_top_level {
        span.metrics.insert(TOP_LEVEL_KEY.into(), 1.0);
    } else {
        span.metrics.remove(TOP_LEVEL_KEY);
    }
}
```

remove works fine if the key doesn't exist https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.remove, just the return value will be None which don't care here.